### ISSUE TITLE
Unifying unit test scenarios

### ISSUE THREAD

**Issue by edeak** [moszinet/BKKCrypt#181]
I checked out some unit tests and they are quite different from each other.
I think we should define the use cases (passwords) which need to be tested in a unit test.
Once we're done with that, new unit tests should follow that, and existing unit tests should be updated.

**Comment by riley-martine** [comment]
What kind of things need to be tested? In the python tests I made sure all the most common passwords and several large dumps all worked, as well as a bunch of tricky strings. I have seen some people making sure it does not encrypt to other strings, like this:
`Assert.AreNotEqual(encryptedPassword,"394257263CF893FE61D72E8E3B1E625F297CDACD5FD2774C0239C4A8A1D4D783F7363B4A20A3848BFD7A7A6833EB858A631CECBE2A297F3AB868111028E4FEF6"); //sha512`

**Comment by edeak** [comment]
Yeah, I think we should collect the scenarios and test them in all implementation.
One scenario is to test the encryption for different inputs. I think the inputs should be the same for all the implementations. **We should have a common agreement on that, which inputs we should use.**

The other scenario as @riley-martine pointed out, that test if the encrypted password not equals to the sha512. We can use any other encryption here to make sure we have no collision with them. **We should have a common agreement on that, which encryptions should we use.**
